main/static/script.js

```diff
@@ -227,15 +198,17 @@ module.controller("mainCtrl", function(
       options.isStacked = true;
       chart.chart = new google.visualization.AreaChart(chart.dom);
     }
-    asyncRender(chart, dataTable, options);
+    $asyncRender($scope, chart, dataTable, options);
   };
+});
 
-  function receiveProfile(object, chart) {
+module.factory("$receiveProfile", function($chartWaiting, $asyncRender) {
```

Comment:
i think both receiveSelect and receiveProfile can be written to _not_ use `$scope` - it's hard to reason around, because `$scope` can contain (virtually) anything. can we pass the required logic as a variable? (also, $scope.apply can be abstracted out to rootscope.$apply and simply be injected).